### ISSUE TITLE
BAD-TITLE: Log steps inside steps

### ISSUE THREAD

**Issue by Tiagojdferreira** [behave/behave#712]
I have been looking at the issues regarding logging and I haven't found anything like this.
I have a test that uses a big step that consistes of smaller steps. For example, imagine I have a step `Then I click the colored buttons` that has a loop for `['red', 'blue', 'green']` where it clicks the button and checks a color changed using `context.execute_steps`.
Currently, when I run the tests I get something like:
```
Scenario: My test  # functional_tests/features/test.feature:10
    Then I click the colored buttons                      # functional_tests/steps/test_steps.py:16
```
And what I would like to have is:
```
Scenario: My test  # functional_tests/features/test.feature:10
    Then I click the colored buttons                      # functional_tests/steps/test_steps.py:16
      When I click the red button                      # functional_tests/steps/test_steps.py:16
      Then I see the red box                      # functional_tests/steps/test_steps.py:16
      When I click the blue button                      # functional_tests/steps/test_steps.py:16
      Then I see the blue box                      # functional_tests/steps/test_steps.py:16
      When I click the green button                      # functional_tests/steps/test_steps.py:16
      Then I see the green box                      # functional_tests/steps/test_steps.py:16
```
And ideally if the test fails in one step, I would get the step where it failed

**Comment by spyoungtech** [comment]
You could consider a [scenario outline](https://jenisys.github.io/behave.example/tutorials/tutorial04.html) for this.


```
Scenario Outline: clicking colored buttons
    When I click the <color> button
    Then I see the <color> box

    Examples: colors
        | color |
        | red   |
        | blue  |
        | green |
```

**Comment by jenisys** [comment]
@Tiagojdferreira 
* Use plain formatter and you will get the passed/failed state of each step
* behave aborts the step execution after the first failed step in a Scenario.
  If only the first step is shown in the Scenario test-run indicate that you have a failure in the first step (w/ the pretty formatter).
* If you use `execute_steps()` and failure occurs in the substeps, this substep is shown as failed step
* If you want to see skipped steps, use `behave --show-skipped` (for example) or 
  provide your own formatter that shows you what want to see.

SEE ALSO:
* [test/test_runner.py:363](https://github.com/behave/behave/blob/master/test/test_runner.py#L363)

**Comment by jenisys** [comment]
By the way: What have all your questions to do with logging ?!?

**Comment by Tiagojdferreira** [comment]
@spyoungtech thanks for the suggestion, but that was a simplified example to show the issue. My original problem comes from getting all routes in my django app and visiting them to see if there is a 404 or a javascript error in it, so I can't really use a scenario outline, as I would have to add each route individually.

@jenisys regarding your points:
`Use plain formatter and you will get the passed/failed state of each step`
Great to know. Is there any way to get them to log even in regular format? Or do I need to add my own formatter for that?

`If you use execute_steps() and failure occurs in the substeps, this substep is shown as failed step`
I noticed this, and that's why I closed the issue. I reopened it because I had another issue but I found a work around. Sorry for not closing it afterwards.

In the end I managed to do what I wanted, but I had to hack a bit around. I think I will open an issue with the details for that problem and close this one.

I put it as logging given that what I was missing was the logs about substeps.
